File: howners-api/src/app/core/services/tenant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { User } from '../models/user.model';
import { Rental } from '../models/rental.model';
import { Contract } from '../models/contract.model';
import { Document } from '../models/document.model';

export interface UpdateProfileRequest {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
}

@Injectable({
  providedIn: 'root'
})
export class TenantService {
  private apiUrl = `${environment.apiUrl}/tenants`;

  constructor(private http: HttpClient) {}

  /**
   * Récupérer le profil du locataire connecté
   */
  getMyProfile(): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/me`);
  }

  /**
   * Mettre à jour le profil du locataire
   */
  updateMyProfile(request: UpdateProfileRequest): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/me`, request);
  }

  /**
   * Récupérer mes locations
   */
  getMyRentals(): Observable<Rental[]> {
    return this.http.get<Rental[]>(`${this.apiUrl}/me/rentals`);
  }

  /**
   * Récupérer mes contrats
   */
  getMyContracts(): Observable<Contract[]> {
    return this.http.get<Contract[]>(`${this.apiUrl}/me/contracts`);
  }

  /**
   * Récupérer mes documents
   */
  getMyDocuments(): Observable<Document[]> {
    return this.http.get<Document[]>(`${this.apiUrl}/me/documents`);
  }
}

File: howners-back/src/main/java/com/howners/gestion/service/tenant/TenantService.java
package com.howners.gestion.service.tenant;

import com.howners.gestion.domain.contract.Contract;
import com.howners.gestion.domain.document.Document;
import com.howners.gestion.domain.rental.Rental;
import com.howners.gestion.domain.user.User;
import com.howners.gestion.dto.request.UpdateProfileRequest;
import com.howners.gestion.dto.contract.ContractResponse;
import com.howners.gestion.dto.document.DocumentResponse;
import com.howners.gestion.dto.response.RentalResponse;
import com.howners.gestion.dto.response.UserResponse;
import com.howners.gestion.exception.ResourceNotFoundException;
import com.howners.gestion.repository.ContractRepository;
import com.howners.gestion.repository.DocumentRepository;
import com.howners.gestion.repository.RentalRepository;
import com.howners.gestion.repository.UserRepository;
import com.howners.gestion.service.auth.AuthService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Slf4j
@Transactional(readOnly = true)
public class TenantService {

    private final UserRepository userRepository;
    private final RentalRepository rentalRepository;
    private final ContractRepository contractRepository;
    private final DocumentRepository documentRepository;

    /**
     * Récupérer le profil du locataire connecté
     */
    public UserResponse getMyProfile() {
        UUID currentUserId = AuthService.getCurrentUserId();
        User user = userRepository.findById(currentUserId)
                .orElseThrow(() -> new ResourceNotFoundException("Utilisateur", "id", currentUserId));
        
        return UserResponse.from(user);
    }

    /**
     * Mettre à jour le profil du locataire
     */
    @Transactional
    public UserResponse updateMyProfile(UpdateProfileRequest request) {
        UUID currentUserId = AuthService.getCurrentUserId();
        User user = userRepository.findById(currentUserId)
                .orElseThrow(() -> new ResourceNotFoundException("Utilisateur", "id", currentUserId));

        // Mise à jour des informations
        user.setFirstName(request.firstName());
        user.setLastName(request.lastName());
        user.setEmail(request.email());
        
        if (request.phone() != null && !request.phone().isBlank()) {
            user.setPhone(request.phone());
        }

        user = userRepository.save(user);
        log.info("Tenant profile updated: {}", user.getEmail());

        return UserResponse.from(user);
    }

    /**
     * Récupérer toutes les locations du locataire connecté
     */
    public List<RentalResponse> getMyRentals() {
        UUID currentUserId = AuthService.getCurrentUserId();
        
        List<Rental> rentals = rentalRepository.findByTenantId(currentUserId);
        log.info("Found {} rentals for tenant {}", rentals.size(), currentUserId);

        return rentals.stream()
                .map(RentalResponse::from)
                .collect(Collectors.toList());
    }

    /**
     * Récupérer tous les contrats du locataire connecté
     */
    public List<ContractResponse> getMyContracts() {
        UUID currentUserId = AuthService.getCurrentUserId();
        
        // Récupérer les locations du locataire
        List<Rental> rentals = rentalRepository.findByTenantId(currentUserId);
        List<UUID> rentalIds = rentals.stream()
                .map(Rental::getId)
                .collect(Collectors.toList());

        if (rentalIds.isEmpty()) {
            return List.of();
        }

        // Récupérer les contrats associés à ces locations
        List<Contract> contracts = contractRepository.findByRentalIdIn(rentalIds);
        log.info("Found {} contracts for tenant {}", contracts.size(), currentUserId);

        return contracts.stream()
                .map(ContractResponse::from)
                .collect(Collectors.toList());
    }

    /**
     * Récupérer tous les documents du locataire connecté
     */
    public List<DocumentResponse> getMyDocuments() {
        UUID currentUserId = AuthService.getCurrentUserId();
        
        List<Document> documents = documentRepository.findByUploaderId(currentUserId);
        log.info("Found {} documents for tenant {}", documents.size(), currentUserId);

        return documents.stream()
                .map(DocumentResponse::from)
                .collect(Collectors.toList());
    }
}
